aten/src/ATen/native/cuda/Loops.cuh

```diff
@@ -127,4 +133,76 @@ void gpu_kernel_with_scalars(TensorIterator& iter, const func_t& f) {
   }
 }
 
+namespace { // functions for `gpu_kernel_multiple_outputs`.
+
+template <int num_outputs, typename func_t, typename array_t, typename inp_calc_t, typename out_calc_t>
+C10_LAUNCH_BOUNDS_1(num_threads)
+__global__ void unrolled_elementwise_kernel_for_multi_outputs(int N, func_t f, array_t data, inp_calc_t ic, out_calc_t oc) {
+  int remaining = N - block_work_size * blockIdx.x;
+  modern::elementwise_kernel_helper(f, memory::policies::multi_outputs_unroll<array_t, inp_calc_t, out_calc_t, num_outputs>(data, remaining, ic, oc));
+}
+
+template <int num_outputs, typename func_t, typename array_t, typename inp_calc_t, typename out_calc_t>
+static inline void launch_unrolled_kernel_for_multi_outputs(int64_t N, const func_t& f, array_t data, inp_calc_t ic, out_calc_t oc) {
+  TORCH_INTERNAL_ASSERT(N > 0 && N <= std::numeric_limits<int32_t>::max());
+  int64_t grid = (N + block_work_size - 1) / block_work_size;
+  auto stream = at::cuda::getCurrentCUDAStream();
+  unrolled_elementwise_kernel_for_multi_outputs<num_outputs, func_t, array_t><<<grid, num_threads, 0, stream>>>(N, f, data, ic, oc);
+  AT_CUDA_CHECK(cudaGetLastError());
+}
+
+template <typename func_t>
+void gpu_kernel_multiple_outputs_impl(TensorIterator& iter, const func_t& f) {
+  using traits = function_traits<func_t>;
+  using output_t = typename traits::result_type;
+  TORCH_INTERNAL_ASSERT(memory::detail::is_tuple<output_t>::value);
+  constexpr int num_outputs = thrust::tuple_size<output_t>::value;
+  TORCH_INTERNAL_ASSERT(num_outputs > 1);
+  constexpr int num_inputs = traits::arity;
```

Comment:
Just a note: this will be changed to `traits::arity()`, depending on when https://github.com/pytorch/pytorch/pull/35774 is landed.